pyright/langserver.py

```diff
@@ -3,13 +3,14 @@
 import sys
 import json
 import tempfile
+from getpass import getuser
 from pathlib import Path
 from typing import Any, NoReturn, Union
 
 from . import node
 
 
-TEMP_DIR = Path(tempfile.gettempdir()) / 'pyright-python-langserver'
+TEMP_DIR = Path(tempfile.gettempdir()) / f'pyright-python-langserver.{getuser()}'
```

Comment:
We should also handle the case where `getuser()` fails, I do not want the CLI to crash if the username cannot be detected.

I would assume it would be safe to then fall back to using the previous locations for the temporary directories as we would not be in a multi-user environment.

Reply:
Good point, fixed.